File: db/event.go
package db

import (
	"time"
)

// Event represents an error that occurs
type Event struct {
	ID        int64      `gorm:"primaryKey" json:"id"`
	Time      *time.Time `gorm:"not null" json:"time"`
	SiteID    int64      `gorm:"not null" json:"site_id"`
	Site      Site       `gorm:"constraint:OnDelete:CASCADE;" json:"-"`
	OldStatus string     `gorm:"not null" json:"old_status"`
	NewStatus string     `gorm:"not null" json:"new_status"`
	Details   string     `gorm:"not null" json:"details"`
}

File: ui/src/types/event.ts
import { z } from 'zod'

export const EventSchema = z.object({
  id: z.number(),
  time: z.string(),
  site_id: z.number(),
  old_status: z.string(),
  new_status: z.string(),
  details: z.string(),
})

export type Event = z.infer<typeof EventSchema>
